go-controller/pkg/ovn/pods.go

```diff
@@ -309,46 +286,44 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	var uuid string
 	uuid, _, err = util.RunOVNNbctl("get", "logical_switch_port", portName, "_uuid")
 	if err != nil {
-		return fmt.Errorf("error while getting UUID for logical port %s "+
-			"stdout: %q, stderr: %q (%v)", portName, uuid, stderr, err)
+		return handleError(logicalSwitch, &podCIDR.IP, addrAssigned, fmt.Errorf("error while getting UUID for logical port %s "+
+			"stdout: %q, stderr: %q (%v)", portName, uuid, stderr, err))
 	}
 	if !strings.Contains(uuid, "-") {
-		return fmt.Errorf("invalid logical port %s uuid %q", portName, uuid)
+		return handleError(logicalSwitch, &podCIDR.IP, addrAssigned, fmt.Errorf("invalid logical port %s uuid %q", portName, uuid))
 	}
 
 	// Add the pod's logical switch port to the port cache
 	portInfo := oc.logicalPortCache.add(logicalSwitch, portName, uuid, podMac, podCIDR.IP)
 
-	// Set the port security for the logical switch port
-	addresses := fmt.Sprintf("%s %s", podMac, podCIDR.IP)
-	out, stderr, err = util.RunOVNNbctl("lsp-set-port-security", portName, addresses)
-	if err != nil {
-		return fmt.Errorf("error while setting port security for logical port %s "+
-			"stdout: %q, stderr: %q (%v)", portName, out, stderr, err)
-	}
-
 	// Enforce the default deny multicast policy
 	if oc.multicastSupport {
 		if err := podAddDefaultDenyMulticastPolicy(portInfo); err != nil {
-			return err
+			return handleError(logicalSwitch, &podCIDR.IP, addrAssigned, err)
 		}
 	}
 
 	if err := oc.addPodToNamespace(pod.Namespace, portInfo); err != nil {
-		return err
+		return handleError(logicalSwitch, &podCIDR.IP, addrAssigned, err)
 	}
 
 	if annotation == nil {
 		hybridOverlayExternalGW := net.IP{}
 		if config.HybridOverlay.Enabled {
 			hybridOverlayExternalGW, err = oc.getHybridOverlayExternalGwAnnotation(pod.Namespace)
 			if err != nil {
-				return err
+				return handleError(logicalSwitch, &podCIDR.IP, addrAssigned, err)
 			}
 		}
+		var nodeSubnet *net.IPNet
+		//FIXME DUAL-STACK
+		if nodeSubnet = oc.lsCache.GetSwitchSubnet(logicalSwitch); nodeSubnet == nil {
+			return fmt.Errorf("Cannot retrieve subnet for assigning gateway routes for pod %s, node: %s", pod.Name, logicalSwitch)
+		}
```

Comment:
L321: you missed `handleError()` here, so you using the defer() funciton is safe.